packages/react-core/src/core/hooks/wallet-hooks.ts

```diff
@@ -0,0 +1,125 @@
+import { useThirdwebWallet } from "../providers/thirdweb-wallet-provider";
+import invariant from "tiny-invariant";
+
+/**
+ * @returns the current active wallet instance
+ */
+export function useActiveWallet() {
+  const context = useThirdwebWallet();
+  invariant(
+    context,
+    "useActiveWallet() hook must be used within a <ThirdwebProvider/>",
+  );
+  return context.activeWallet;
+}
+
+/**
+ *
+ * @returns `supportedWallets` configured in the `<ThirdwebProvider/>`
+ */
+export function useWallets() {
+  const context = useThirdwebWallet();
+  invariant(
+    context,
+    "useWallets() hook must be used within a <ThirdwebProvider/>",
+  );
+  return context.wallets;
+}
+
+/**
+ *
+ * @returns a method to connect to a wallet class
+ */
+export function useConnect() {
+  const context = useThirdwebWallet();
+  invariant(
+    context,
+    "useConnect() hook must be used within a <ThirdwebProvider/>",
+  );
+  return context.connect;
+}
+
+/**
+ *
+ * @returns a method to disconnect from the current active wallet
+ */
+export function useDisconnect() {
+  const context = useThirdwebWallet();
+  invariant(
+    context,
+    "useDisconnect() hook must be used within a <ThirdwebProvider/>",
+  );
+  return context.disconnect;
+}
+
+/**
+ *
+ * @returns the connection status of the wallet
+ *
+ * It can be one of the following:
+ * 1. `unknown` - when wallet connection status is not yet known
```

Comment:
what causes `unknown`? just a loading state? can we make it `loading` in that case?

Reply:
status starts out with `unknown` - because when the app is loading, we just don't know status

Here are two possible scenarios when app is loading:

1. if wallet was previously connected + autoconnect enabled, the status goes from `unknown` => `connecting` => `connected`
2. if no wallet was connected before or autoconnect is not enabled, the status goes from `unknown` => `disconnected`


We can make it `loading` instead of `unknown` but it sounds a bit similar to `connecting` - it feels like we are trying to "connect" but we are not. Let me know what you think, I'm fine with either